include/TaskContext.hpp

```diff
@@ -0,0 +1,91 @@
+#ifndef TASK_CONTEXT_HPP
+#define TASK_CONTEXT_HPP
+
+#include <freertos/FreeRTOS.h>
+#include <freertos/semphr.h>
+#include <freertos/task.h>
+
+#include "OperationMode.hpp"
+
+// forward declaration
+class Hardware;
```

Comment:
Is the purpose of forwarding the Hadware class is to not having access to its properties in TaskContext?

Reply:
It's a forward declaration, a useful trick in C/C++ to avoid circular `#include`s and too much bloat.